skopt/tests/test_parameter.py

```diff
@@ -0,0 +1,86 @@
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_equal
+
+from skopt.parameter import sample_points
+from skopt.parameter import Uniform
+from skopt.parameter import Integer
+from skopt.parameter import Categorical
+
+
+def check_distribution(Dist, vals, random_val):
+    x = Dist(*vals)
+    assert_equal(x.rvs(random_state=1), random_val)
+
+
+def test_distributions():
+    yield (check_distribution, Uniform, (1., 4.), 2.668088018810296)
+    yield (check_distribution, Uniform, (1, 4), 2.668088018810296)
+    yield (check_distribution, Integer, (1, 4), 2)
+    yield (check_distribution, Integer, (1., 4.), 2)
+    yield (check_distribution, Categorical, ('a', 'b', 'c', 'd'), 'b')
+    yield (check_distribution, Categorical, (1., 2., 3., 4.), 2.)
+
+
+def test_simple_grid():
+    expected = [(2, 1), (1, 2), (2, 2), (2, 1), (1, 2)]
+
+    for i,p in enumerate(sample_points([(1, 3), (1, 4)],
+                                       len(expected), random_state=1)):
+        assert_equal(p, expected[i])
+
+
+def test_simple_categorical():
+    expected = [(2, 1), (1, 2), (2, 2), (2, 1), (1, 2)]
+
+    for i,p in enumerate(sample_points([Categorical(1, 2), (1, 4)],
+                                       len(expected), random_state=1)):
+        assert_equal(p, expected[i])
```

Comment:
You can maybe increase the length of the interval in the above and check that
1. Every value lies within the bounds if the length is equal to 2
2. Every value is always in each tuple, if Categorical is explicitly provided or length of each tuple is greater than 2.